interpreter/interpreter.go

```diff
@@ -11,10 +11,11 @@ import (
 )
 
 type Interpreter struct {
-	types       map[semantic.Node]semantic.Type
-	polyTypes   map[semantic.Node]semantic.PolyType
-	sideEffects []values.Value
-	pkg         string
+	types             map[semantic.Node]semantic.Type
+	polyTypes         map[semantic.Node]semantic.PolyType
+	sideEffects       []SideEffect // a list of the side effects occurred during the last call to `Eval`.
+	globalSideEffects []SideEffect // a list of every side effect occurred within multiple evaluations.
```

Comment:
Is this field necessary? Only the repl calls Eval multiple times, but I believe it only cares about the side effects produced from the last call.

Reply:
That's fine, I wanted to preserve somehow the previous behavior.